example/sinatra/index.erb

```diff
@@ -1,214 +1,189 @@
-  Veritrans.client_key = "<%= Veritrans.config.client_key %>";
-
-  function createTokenData() {
-    return {
-      // Optional params:
-      // secure: true
-      // bank: 'MANDIRI'
-
-      card_number: $('#card_number').val(),
-      card_cvv: $('#card_cvc').val(),
-      card_exp_month: $('#card_exp').val().match(/(\d+) \//)[1],
-      card_exp_year: '20' + $('#card_exp').val().match(/\/ (\d+)/)[1],
-      gross_amount: $('#gross_amount').val(),
-      secure: $('#card_secure')[0].checked
-    };
-  }
-
-  $(document).ready(function () {
-    $('#card_number').payment('formatCardNumber');
-    $('#card_cvc').payment('formatCardCVC');
-    $('#card_exp').payment('formatCardExpiry');
-
-    $('#card_form').on('submit', function (event) {
-      var form = this;
-      $('#submit_btn').attr('disabled', true).val("Processing ...");
-      event.preventDefault();
-
-      Veritrans.token(createTokenData, function (data) {
-        console.log('Token data:', data);
-        // if we get url then it's 3d-secure transaction
-        // so we need to open that page
-        // this callback function will be called again after user confirm 3d-secure
-        // you can also redirect on server side
-        if (data.redirect_url) {
-          $.magnificPopup.open({
-            items: { type: 'iframe', src: data.redirect_url }
-          });
-          $.magnificPopup.instance.content.find('iframe').height(590);
-        // if no redirect_url and we have token_id then just make charge request
-        } else if (data.token_id) {
-          $('#card_token').val(data.token_id);
-          $.magnificPopup.close();
-          form.submit();
-        // if no redirect_url and no token_id, then it should be error
-        } else {
-          alert(data.validation_messages ? data.validation_messages.join("\n") : data.status_message);
-          $('#submit_btn').removeAttr('disabled').removeAttr("value");
+    $(function () {
+        // open the console log to check the flow
+        // 3ds new flow:
+        // 1. get token_id
+        // 2. send token_id to backend
+        // 3. initial charge from backend to midtrans api
+        // 4. open redirect_url
+
+        var options = {
+            performAuthentication: function(redirect_url){
+                openDialog(redirect_url);
+            },
+            onSuccess: function(response){
+                console.log('success');
+                console.log('response:',response);
+                closeDialog();
+            },
+            onFailure: function(response){
+                console.log('fail');
+                console.log('response:',response);
+                closeDialog();
+                alert(response.status_message);
+                $('button').removeAttr("disabled");
+            },
+            onPending: function(response){
+                console.log('pending');
+                console.log('response:',response);
+                closeDialog();
+            }
+        };
```

Comment:
The demo feels incomplete, why only output to console.log? can't seems to attempt another payment after success even. Can you update the example to be like the other library?
https://github.com/Xaxxis/midtrans-go/blob/e3b1ef0613036dc224ed59517e3aeb57d2f2edcb/example/simple/coreapi-card-3ds/views/index.html#L140-L156
or
https://github.com/Midtrans/midtrans-nodejs-client/blob/b25bfd2e7364900893d515b24f3768c5fe1ea1c1/examples/expressApp/views/simple_core_api_checkout.ejs#L121-L138

Reply:
also please rename the `sinatra.rb` into app.rb or webapp.rb, because sinatra is already on the folder name.